RealexPayments/HPP/Controller/Process/Result/Base.php

```diff
@@ -148,13 +182,16 @@ private function _validateResponse($response)
         $result = $response['RESULT'];
         $orderid = $response['ORDER_ID'];
         $message = $response['MESSAGE'];
-        $authcode = $response['AUTHCODE'];
         $pasref = $response['PASREF'];
         $realexsha1 = $response['SHA1HASH'];
 
         $merchantid = $this->_helper->getConfigData('merchant_id');
 
-        $sha1hash = $this->_helper->signFields("$timestamp.$merchantid.$orderid.$result.$message.$pasref.$authcode");
+        if ($this->_paymentManagement->isTransactionApm($response)) {
+            $sha1hash = $this->_helper->signFields("$timestamp.$merchantid.$orderid.$result.$message.$pasref.");
+        } else {
+            $sha1hash = $this->_helper->signFields("$timestamp.$merchantid.$orderid.$result.$message.$pasref.{$response['AUTHCODE']}");
+        }
```

Comment:
These formats are the same with the exception of attempting to add the auth code. This can be simplified back to the original form by changing how `$authcode` is assigned, e.g.:

```php
$authcode = $response['AUTHCODE'] ?? '';
```

Reply:
The response hash documented at https://developer.globalpay.com/hpp/payment-methods#hpp-return-url contains `paymentmethod` as the last component of the string to be hashed. Found out it got removed in the meantime, but I would still keep them separated so it's clear/transparent that the string to hash is different, despite the fact that the only difference between them is the missing `authcode` for now.